src/ray/gcs/tables.h

```diff
@@ -223,6 +226,10 @@ class Table : private Log<ID, Data>, public TableInterface<ID, Data> {
   Status Add(const JobID &job_id, const ID &id, std::shared_ptr<DataT> data,
              const WriteCallback &done);
 
+  /// A general version of Add() that supports different CommandType's.
+  Status Add(const JobID &job_id, const ID &id, std::shared_ptr<DataT> data,
```

Comment:
I don't think it makes sense to expose the command type in the `Table` interface. Then everyone who calls operations on a table will have to think about whether the table is chain-replicated or not. And if both versions of a table are supported, like for the task table, then there will be no error if we try to `Add` to the chain-replicated version in one place and `Add` to the non-chain-replicated version in another. Can you hide the `CommandType` by moving it into the implementation (e.g., set it to `kRegular` in the default `Table` constructor and set it to `kChain` in the task table derived classes)?

Reply:
I went ahead and investigated per our offline discussion.  Here's another thought.  If we make `kRegular/kChain` a ctor arg for `AsyncGcsClient` and various `Table`s,  it gives the reader a hint that _all_ commands with such a client/table pair will be chain-enabled.  Which is not the case; therefore I think incrementally adding such support to what's actually been enabled, e.g. `Table::Add(..., CommandType _)`, makes more sense.  

What do you think?

Reply:
I still don't think it makes sense to expose it in the `Add` interface. From the caller's perspective, I think it's clearest if we make the granularity for chain replication at the level of a table, not a command. For instance, I don't really understand what it means for `Add` to be supported but not `Lookup`. If your concern is about misleading the caller, I would just add a check on any unsupported methods that the `CommandType` flag is set to `kRegular`.

Reply:
"For instance, I don't really understand what it means for Add to be supported but not Lookup" <- this is a great argument & feedback.  I agree.  Let me see what I can do.